sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/javaCode.scala

```diff
@@ -148,14 +149,17 @@ trait Block extends JavaCode {
   }
 
   // Concatenates this block with other block.
-  def + (other: Block): Block
+  def + (other: Block): Block = other match {
```

Comment:
A general question about `+`.

Previously we generate a giant string for an expression tree, which is hard to tune. To keep more information in the generated code, we introduce this `JavaCode`/`CodeBlock` framework to keep a tree of strings instead of a giant string.

For an expression `a op b`, we should generate a tree of strings for `a` and `b`, then `op` creates a new tree node and keeps `a` and `b` as children. That means, if we refer to a `CodeBlock` inside a `code"..."`, the `CodeBlock` should become a child of the new `CodeBlock`. However, `+` usually happens within the same operator, I'm not sure if we should create a new level of tree node here.

Reply:
I think the tree structure itself is an important information in the generated code, we should think carefully about what a tree node means. For example, when we want to split the code into methods, how shall we deal with the tree node? Shall we split an entire tree node into one method? What's our assumption to the java code inside one tree node?

Reply:
The ideal usage is that we all put a semantically integral java code into a `CodeBlock`. If the `CodeBlock` is produced by an expression in codegen, in order to split it, we should split the entire tree of the `CodeBlock`.

Reply:
that's a good answer, let's make sure that when we call `+`, the 2 blocks are 2 individual semantically integral java code.